e2e/restore_test.go

```diff
@@ -0,0 +1,109 @@
+package e2e
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/NethermindEth/eigenlayer/internal/common"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRestore(t *testing.T) {
+	// Test context
+	var (
+		backupId   string
+		restoreErr error
+	)
+	// Build test case
+	e2eTest := newE2ETestCase(
+		t,
+		// Arrange
+		func(t *testing.T, eigenlayerPath string) error {
+			// Build option returner image
+			err := buildOptionReturnerImageLatest(t)
+			if err != nil {
+				return err
+			}
+			// Install option returner AVS
+			err = runCommand(t, eigenlayerPath, "install", "--profile", "option-returner", "--no-prompt", "--yes", "--version", common.MockAvsPkg.Version(), "--option.test-option-hidden", "12345678", "--option.test-option-enum-hidden", "option3", common.MockAvsPkg.Repo())
+			if err != nil {
+				return err
+			}
+			// Backup AVS
+			backupOut, err := runCommandOutput(t, eigenlayerPath, "backup", "mock-avs-default")
+			if err != nil {
+				return err
+			}
+			// Parse backup id
+			r := regexp.MustCompile(`.*Backup created with id: (?P<backup_id>[a-f0-9]+).*`)
+			matches := r.FindSubmatch(backupOut)
+			require.Len(t, matches, 2)
+			backupId = string(matches[1])
+			// Uninstall AVS
+			return runCommand(t, eigenlayerPath, "uninstall", "mock-avs-default")
+		},
+		// Act
+		func(t *testing.T, egnPath string) {
+			restoreErr = runCommand(t, egnPath, "restore", backupId)
+		},
+		// Assert
+		func(t *testing.T) {
+			assert.NoError(t, restoreErr, "restore command should not fail")
+			checkInstanceExists(t, "mock-avs-default")
```

Comment:
As far as I know, the option returner does not have an endpoint to change an option value, only to change the health status for testing purposes. We can implement this scenario, but we need to add the following features to the option-returner profile:

- Save option values in a file and bind the file as a volume to ensure it is saved in the backup. Currently, we are accessing options as environment variables declared in the docker compose, which are in-memory values that will not be saved during the backup process.
- Create a new endpoint, POST `/option/update?option={option_name}&value={option_value}`, to change an option value by updating the file.
- Update the `/option/{option-name}` endpoint to read from the file instead of the environment variables.

As for checking the version, commit, profile, etc., I can save the `state.json` file before the backup and compare it to the new one after the restore.

Reply:
> As for checking the version, commit, profile, etc., I can save the state.json file before the backup and compare it to the new one after the restore.

Yes please.

Reply:
You are right, we don't have such endpoint yet but I'm thinking it may not be necessary after all.

Given this:
> Currently, we are accessing options as environment variables declared in the docker compose, which are in-memory values that will not be saved during the backup process.

I have been thinking. Those options are env variables stored in the `.env` file -which is back up-, then after restoring that backup the compose project initializes with the env variables from the backed up `.env` file right? This means we can install the instance initially using some custom values for some options (`--option.<option_name>`) and after restoring the instance query the API and check that those values are the same as for the backed up instance. This doesn't guarantee us that it is the same instance, but this check along with the `state.json` info, puts a pretty good foundation.

Reply:
I agree, the .env file is sufficient

Reply:
Done https://github.com/NethermindEth/eigenlayer/pull/90/commits/81e4b7e8dc5422cc45102dd20cefe687be65aed9